skiplist/src/list.rs

```diff
@@ -431,8 +444,20 @@ impl<T: AsRef<Skiplist<C>>, C: KeyComparator> IterRef<T, C> {
 
     pub fn prev(&mut self) {
         assert!(self.valid());
-        unsafe {
-            self.cursor = self.list.as_ref().find_near(self.key(), true, false);
+        if self.list.as_ref().allow_concurrent_write {
+            unsafe {
+                self.cursor = self.list.as_ref().find_near(self.key(), true, false);
+            }
```

Comment:
Can we apply this improvement when concurrent write is allowed?

Reply:
It is much more complicated for concurrent write, ref to https://github.com/tikv/rocksdb/pull/278/files, maybe in the future we can port it.

Reply:
I see, exactly not easy :joy: